visualization/app/codeCharta/ui/mapTreeView/mapTreeView.level.po.ts

```diff
@@ -3,12 +3,19 @@ import { Page } from "puppeteer"
 export class MapTreeViewLevelPageObject {
 	constructor(private page: Page) {}
 
-	public async openContextMenu() {
-		await this.page.click("[id='/root/ParentLeaf']")
-		await this.page.click("[id='/root/ParentLeaf/smallLeaf.html']", { button: "right" })
+	public async openContextMenu(path: string) {
+		await this.page.click(`[id='${path}'] > div`, { button: "right" })
 	}
 
-	public async nodeExists() {
-		return !!(await this.page.$("[id='/root/ParentLeaf/smallLeaf.html']"))
+	public async openFolder(path: string) {
+		await this.page.click(`[id='${path}'] > div`)
+	}
+
+	public async hoverNode(path: string) {
+		await this.page.hover(`[id='${path}'] > div`)
+	}
+
+	public async nodeExists(path: string) {
+		return !!(await this.page.$(`[id='${path}'] > div`))
```

Comment:
Is the `> div` part actually necessary? I would expect an ID to be unique?

Reply:
It was necessary, when clicking on folders. But I moved the id to the inner div, so that we don't need that anymore.
The inner div is where the click listeners are located anyway. So it all makes sense now 👍